enhancements/kubelet/virtualization-higher-workload-density.md

```diff
@@ -0,0 +1,474 @@
+  proposal, since enabling swap on control-plane nodes is not supported.
+* HCP: Hosted cluster with OCP Virt, since nested virt is not supported.
+
+## Proposal
+
+A higher workload density is achieved by combining two mechanisms
+
+1. Under-request memory resources for virtual machines by having the VM
+   pods (`launcher`) `requests.memory` being smaller than the VMs `guest.memory`
+   according to a configured ratio[^1]. This is owned by
+   [KubeVirt and present today](https://kubevirt.io/user-guide/operations/node_overcommit/).
+2. Compensate for the memory over-committment by using SWAP in order
+   to extend the virtual memory. This is owned by the platform and not
+   available today.
+
+
+### Applicability
+
+We expect to mitigate the following situations
+
+* Underutilized guest memory (as described above)
+* Infrastructure related memory spikes in an over-committed environment
+* Considerable amount of "cold memory allocations" - thus allocations
+  which are rarely used
+
+### Scope
+
+Following the upstream kubernetes approach every workload marked as burstable QoS would be able to swap.
+There is no differentiation between the type of the workload: regular pod or a VM.
+With that being said, swapping will be allowed by WASP (and later on by kube swap) for pods
+with Burstable QoS class.
+
+Among the VM workloads, VMs of high-performance configuration (NUMA affinity, CPU affinity, etc.) cannot overcommit.
+Also, VMs with best-effort QoS class don't exist because requesting memory is mandatory for the VM spec. 
+For VMs of Burstable Qos Class over-commited VM stability can be achieved during memory spikes by swapping out "cold" memory pages.
+
+### Timeline & Phases
+
+| Phase                                                                          | Target       |   
+|--------------------------------------------------------------------------------|--------------|
+| Phase 1 - Out-Of-Tree SWAP with WASP TP                                        | mid 2024     |
+| Phase 2 - WASP GA with swap-based evictions                                    | end 2024     |
+| Phase 3 - Transition to swap operator. Limited to CNV-only users               | mid-end 2025 |
+| Phase 4 - Limits removed. All Openshift users allowed to use the swap operator | TBD          |
+
+* **Phase 1 (Done)** - OpenShift Virtualization will provide an out-of-tree
+  solution (WASP) to enable higher workload density and swap-based eviction mechanisms.
+* **Phase 2 (Done)** - WASP will include swap-based eviction mechanisms.
+* **Phase 3 (In Progress)** - OpenShift Virtualization will transition to swap operator.
+  OpenShift will [allow](#swap-ga-for-cnv-users-only) SWAP to be configured only if the swap operator is installed on the cluster.
+* **Phase 4 (TBD)** - OpenShift will allow every user to opt-in for swap. Swap operator will become stand-alone operator deployed by OLM.
+
+### Swap Operator
```

Comment:
Can we reorder the timeline and introduce this section? Otherwise one would have to come down here to understand how swap operator works first.